sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/models/ShareEnabledProtocols.java

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share.models;
+
+import com.azure.storage.common.implementation.Constants;
+
+/**
+ * Represents protocols that can be set on a share.
+ */
+public class ShareEnabledProtocols {
+
+    private boolean smb;
+
+    private boolean nfs;
+
+    /**
+     * @return Enable SMB
+     */
+    public boolean isSmb() {
+        return smb;
+    }
+
+    /**
+     * @return Enable NFS
+     */
+    public boolean isNfs() {
+        return nfs;
+    }
+
+    /**
+     * @param smb Enable SMB
+     * @return The updated object
+     */
+    public ShareEnabledProtocols setSmb(boolean smb) {
+        this.smb = smb;
+        return this;
+    }
+
+    /**
+     * @param nfs Enable NFS
+     * @return The updated object
+     */
+    public ShareEnabledProtocols setNfs(boolean nfs) {
+        this.nfs = nfs;
+        return this;
+    }
+
+    /**
+     * Converts the given protocols to a {@code String}.
+     *
+     * @return A {@code String} which represents the enabled protocols.
+     * @throws IllegalArgumentException If both SMB and NFS are set.
+     */
+    public String toString() {
+        if (this.smb) {
+            if (this.nfs) {
+                throw new IllegalArgumentException("SMB and NFS cannot both be set.");
+            }
+            return Constants.HeaderConstants.SMB_PROTOCOL;
+        }
+        if (this.nfs) {
+            return Constants.HeaderConstants.NFS_PROTOCOL;
+        }
+        return null;
+    }
+
+    /**
+     * Parses a {@code String} into a {@code SharEnabledProtocol}.
+     *
+     * @param str The string to parse.
+     * @return A {@code ShareEnabledProtocol} represented by the string.
+     * @throws IllegalArgumentException If the String is not a recognized protocol.
+     */
+    public static ShareEnabledProtocols parse(String str) {
+        if (str == null) {
+            return null;
+        }
+
+        if (str.equals(Constants.HeaderConstants.SMB_PROTOCOL)) {
+            return new ShareEnabledProtocols().setSmb(true);
+        } else if (str.equals(Constants.HeaderConstants.NFS_PROTOCOL)) {
+            return new ShareEnabledProtocols().setNfs(true);
+        }
+
+        throw new IllegalArgumentException("String is not an understood protocol: " + str);
```

Comment:
@rickle-msft FYI this discussion here is still developing https://github.com/Azure/azure-sdk-for-net/pull/16123#discussion_r518945485 . We might need to loosen this validation if it turns out that new protocols can be returned with current service version.

Reply:
Maybe this should be `ExpandableStringEnum` like this thing https://github.com/Azure/azure-sdk-for-java/blob/7e6e16499bd02077e9159245ce5b43e43ebf9af7/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AccessTier.java#L14

Reply:
I don't think ExpandableStringEnum lets you set more than one?

Reply:
That would bring us back to ExpandableStringEnum<ShareProtocol> and Set<ShareProtocol> (a variation of enumset). Question is if we want to expose future expansion to older SDK users or not.

Btw. Let's not block on this here as we are yet to hear from the service team. we can address this later.